code/lib/core-events/src/errors/preview-errors.ts

```diff
@@ -48,3 +48,161 @@ export class MissingStoryAfterHmrError extends StorybookError {
     - Also check the browser console and terminal for potential error messages.`;
   }
 }
+
+export class MissingStoryFromCsfFileError extends StorybookError {
+  readonly category = Category.PREVIEW_API;
+
+  readonly code = 2;
+
+  constructor(public data: { storyId: string }) {
+    super();
+  }
+
+  template() {
+    return dedent`
+    Couldn't find story matching id '${this.data.storyId}' after importing a CSF file.
+
+    The file was indexed as if the story was there, but then after importing the file in the browser
+    we didn't find the story. Possible reasons:
+    - You are using a custom story indexer that is misbehaving.
+    - You have a custom file loader that is removing or renaming exports.
+
+    Please check your browser console and terminal for errors that may explain the issue.`;
+  }
+}
+
+export class CalledExtractOnStoreError extends StorybookError {
+  readonly category = Category.PREVIEW_API;
+
+  readonly code = 3;
+
+  constructor(public data: object) {
+    super();
+  }
+
+  template() {
+    return dedent`
+    Cannot call \`storyStore.extract()\` without calling \`storyStore.cacheAllCsfFiles()\` first.
+
+    You probably meant to call \`await preview.extract()\` which does the above for you.`;
+  }
+}
+
+export class MissingRenderToCanvasError extends StorybookError {
+  readonly category = Category.PREVIEW_API;
+
+  readonly code = 4;
+
+  constructor(public data: object) {
+    super();
+  }
+
+  template() {
+    return dedent`
+      Expected your framework's preset to export a \`renderToCanvas\` field.
+
+      Perhaps it needs to be upgraded for Storybook 6.4?
+
+      More info: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mainjs-framework-field`;
+  }
+}
```

Comment:
A few observations:
- you can pass a link to the `documentation` property and it will properly format it for you
- you don't need a constructor if the error doesn't take any data. If it does and you do have the constructor, the data will also be passed to telemetry

```suggestion
export class MissingRenderToCanvasError extends StorybookError {
  readonly category = Category.PREVIEW_API;

  readonly code = 4;
  
  readonly documentation =
    'https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mainjs-framework-field';

  template() {
    return dedent`
      Expected your framework's preset to export a \`renderToCanvas\` field.

      Perhaps it needs to be upgraded for Storybook 6.4?`;
  }
}
```